docs/source/caching.mdx

```diff
@@ -177,3 +177,31 @@ store.withinReadWriteTransaction({ transaction in
   }
 })
 ```
+
+### delete
+
+Delete functionality is limited at this time. There are presently two deletion methods available on `ApolloStore`, which are supported by both the in memory and the `SQLite` caches:
+
+1. `clear` - Removes everything from the cache immediately. This is basically the "Nuke it from orbit, it's the only way to be sure" option. 
+2. `removeObject(for:)` on `ReadWriteTransaction`. Removes a single object for the given `CacheKey`. 
+
+`removeObject(for:)` will only remove things at the level of an object - that is, it cannot remove individual properties from an object, and it cannot remove outside references **to** an object. 
+
+You will need to have an understanding of how you are generating cache keys to be able to use this method effectively. If you're taking advantage of our `cacheKeyForObject` function, that will help significantly: You'll have a clear understanding of how cache keys are generated, so you can easily figure out how to construct a cache key to delete. 
+
+For instance, let's say your `cacheKeyForObject` function looks like this: 
+
+```swift
+apollo.cacheKeyForObject = { $0["id"] }
+```
+
+This would indicate that for every object which has an `id` property, it will be cached with that `id` as the key. This would be ideal for an API which uses globally unique identifiers, as our Star Wars API does. 
+
+This means that if you have previously fetched an object with the ID `"2001"`, you can call `transaction.removeObject(for: "2001")` on a `ReadWriteTransaction` to remove any object cached with that key, along with all of its associated scalar properties and the references to other objects it stores. 
+
+`removeObject` does _not_ cascade its removals - if you remove an object which has reference to another object, the reference will be removed, but that other object will not be removed and will remain in the cache. Likewise, if you delete an object, references _to_ that object will not be deleted, they will simply fail when you attempt to load the object again. 
```

Comment:
Maybe mention that depending on the cache policy, failure because of dangling references doesn't necessary imply an error but will just result in a cache miss and refetching the data from the server.

Reply:
yeah I think I can rephrase this to "to fail, causing a cache miss," which should theoretically clarify that a bit more.